nyaa/api_torznab.py

```diff
@@ -101,71 +112,113 @@ def torznab_caps():
     return render_torznab_caps()
 
 
-def torznab_search():
-    page_size = app.config['RESULTS_PER_PAGE']
+def get_categories_filter(cat):
+    categories = []
+    if cat:
+        for item in cat.split(','):
+            # Parse 1_1 format for compatibility
+            cat_match = re.match(r'^(\d+)_(\d+)$', item)
+            if cat_match:
+                item = int(cat_match.group(1)) * 100 + int(cat_match.group(2))
+
+            num = int(item)
+            if num >= 100000:
+                categories.append(num - 100000)
+            elif num >= 1000:
+                categories.extend([x - 100000 for x in TORZNAB_CATEGORY_MAP.get(num, [])])
```

Comment:
`categories` should be extended with a generator expression instead of a list comprehension to reduce the memory requirements of building an unnecessary list.  You can either do it directly in the extend as with your list comp, or create a genexpr variable and extend categories with it:
```python
categories.extend((x - 100000 for x in TORZNAB_CATEGORY_MAP.get(num, [])))
# OR
cat_map = (x - 100000 for x in TORZNAB_CATEGORY_MAP.get(num, []))
categories.extend(cat_map)
```